packages/replayio/src/utils/spawnProcess.ts

```diff
@@ -29,30 +34,28 @@ export function spawnProcess(
     // github.com/replayio/replay-cli/pull/344#discussion_r1553258356
     spawned.unref();
   } else {
+    const stderr = collectData(spawned.stderr);
+
     spawned.on("error", error => {
       deferred.rejectIfPending(error);
-
-      if (onError) {
-        onError(error);
-      }
     });
 
     spawned.on("spawn", () => {
-      if (onSpawn) {
-        onSpawn();
-      }
+      onSpawn?.();
     });
 
     spawned.on("exit", (code, signal) => {
-      if (code || signal) {
-        deferred.rejectIfPending(new Error(`Process failed (code: ${code}, signal: ${signal})`));
-      } else {
-        deferred.resolveIfPending();
-      }
-
-      if (onExit) {
-        onExit();
+      if (code) {
+        const buffered = Buffer.concat(stderr).toString();
+
+        let message = `Process failed (code: ${code})`;
+        if (buffered.length) {
+          message += `:\n${buffered}`;
+        }
+        deferred.rejectIfPending(new Error(message));
+        return;
       }
+      deferred.resolveIfPending();
```

Comment:
Nit (feel free to ignore)
```suggestion
      } else {
        deferred.resolveIfPending();
      }
```